lang/java/reef-bridge-proto-java/src/main/java/org/apache/reef/bridge/driver/service/grpc/GRPCDriverService.java

```diff
@@ -0,0 +1,1084 @@
+   */
+  private boolean driverProcessIsAlive() {
+    if (this.driverProcess != null) {
+      try {
+        this.driverProcess.exitValue();
+      } catch (IllegalThreadStateException e) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private EvaluatorDescriptorInfo toEvaluatorDescriptorInfo(final EvaluatorDescriptor descriptor) {
+    if (descriptor == null) {
+      return null;
+    } else {
+      return EvaluatorDescriptorInfo.newBuilder()
+          .setCores(descriptor.getNumberOfCores())
+          .setMemory(descriptor.getMemory())
+          .setRuntimeName(descriptor.getRuntimeName())
+          .build();
+    }
+  }
+
+  @Override
+  public IdleMessage getIdleStatus() {
+    final String componentName = "Java Bridge DriverService";
+    if (this.clientStub != null) {
+      try {
+        final IdleStatus idleStatus = this.clientStub.idlenessCheckHandler(VOID).get();
+        LOG.log(Level.INFO, "is idle: " + idleStatus.getIsIdle());
+        return new IdleMessage(
+            componentName,
+            idleStatus.getReason(),
+            idleStatus.getIsIdle());
+      } catch (ExecutionException | InterruptedException e) {
+        stop(e);
+      }
+    }
+    return new IdleMessage(
+        componentName,
+        "stub not initialized",
+        true);
+  }
+
+  @Override
+  public void startHandler(final StartTime startTime) {
+    try {
+      start();
+      synchronized (this) {
+        if (this.clientStub != null) {
+          this.clientStub.startHandler(
+              StartTimeInfo.newBuilder().setStartTime(startTime.getTimestamp()).build());
+        } else {
+          stop(new IllegalStateException("Unable to start driver client"));
+        }
+      }
+    } catch (IOException | InterruptedException e) {
+      stop(e);
+    }
+  }
+
+  @Override
+  public void stopHandler(final StopTime stopTime) {
+    synchronized (this) {
+      if (clientStub != null) {
+        final Future<ExceptionInfo> callCompletion = this.clientStub.stopHandler(
+            StopTimeInfo.newBuilder().setStopTime(stopTime.getTimestamp()).build());
+        try {
+          try {
+            final ExceptionInfo error = callCompletion.get(5L, TimeUnit.MINUTES);
+            if (!error.getNoError()) {
+              final Optional<Throwable> t = parseException(error);
+              if (t.isPresent()) {
+                throw new RuntimeException("driver stop exception",
+                    t.get().getCause() != null ? t.get().getCause() : t.get());
+              } else {
+                throw new RuntimeException(error.getMessage() != null ? error.getMessage() : error.getName());
+              }
+            }
+          } catch (TimeoutException e) {
+            throw new RuntimeException("stop handler timed out", e);
+          }
+        } catch (InterruptedException | ExecutionException e) {
+          throw new RuntimeException(e);
+        } finally {
```

Comment:
remove one of the `try` statements and flatten out the `catch` part, i.e.
```java
try {
  // ...
} catch (final TimeoutException e) {
  throw new RuntimeException("stop handler timed out", e);
} catch (final InterruptedException | ExecutionException e) {
  throw new RuntimeException("error in stop handler", e);
} finally {
  stop();
}
```
(also - make exceptions `final` and add proper message to each `RuntimeException`)